src/hooks/use-assistant.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+interface Message {
+	content: string;
+	role: 'user' | 'assistant';
+}
+
+export const useAssistant = ( selectedSiteId: string ) => {
+	const [ messages, setMessages ] = useState< Message[] >( [] );
+
+	useEffect( () => {
+		const storedChat = localStorage.getItem( selectedSiteId );
+		if ( storedChat ) {
+			setMessages( JSON.parse( storedChat ) );
+		} else {
+			localStorage.setItem( selectedSiteId, JSON.stringify( [] ) );
+		}
+	}, [ selectedSiteId ] );
+
+	const addMessage = ( content: string, role: 'user' | 'assistant' ) => {
+		setMessages( ( prevMessages ) => {
+			const updatedMessages = [ ...prevMessages, { content, role } ];
+			localStorage.setItem( selectedSiteId, JSON.stringify( updatedMessages ) );
```

Comment:
Thanks for the feedback on this section. This initial hook was intended to be iterative, but I agree we should explore avoiding the side effects from the updater function and local storage syncing. To reduce development complexity, I think it may be worthwhile to wait until implementing some of the API calls (at least partially), or remove some of the simulated server responses that were included here to more closely match the ultimate intended experience.